tests/Unit/ParallelAlgorithms/Interpolation/Test_ObserveTimeSeriesAndSurfaceData.cpp

```diff
@@ -815,27 +835,44 @@ SPECTRE_TEST_CASE(
         // The interpolation is not perfect because I use too few grid points.
         Approx custom_approx = Approx::custom().epsilon(1.e-4).scale(1.0);
         for (size_t i = 0; i < expected_integral.size(); ++i) {
-          CHECK(expected_integral[i] == custom_approx(written_data(0, i + 1)));
+          if constexpr (ValidPoints ==
+                        InterpTargetTestHelpers::ValidPoints::None) {
+            CHECK_THAT(written_data(0, i + 1), Catch::Matchers::IsNaN());
+          } else {
+            CHECK(expected_integral[i] ==
+                  custom_approx(written_data(0, i + 1)));
+          }
         }
       };
   check_file_contents(expected_integral_a, expected_legend_a, "/SurfaceA");
   check_file_contents(expected_integral_b, expected_legend_b, "/SurfaceB");
   check_file_contents(expected_integral_c, expected_legend_c, "/SurfaceC");
 
   // Check that the Ylm data were written correctly
+  // As this data depends only on the known target (a KerrHorizon) it
+  // uses no interpolated data
   check_ylm_data(h5_file_name);
 
   if (file_system::check_if_file_exists(h5_file_name)) {
     file_system::rm(h5_file_name, true);
   }
 
   // Check that the Surfaces file contains the correct surface data
-  check_surface_volume_data(surfaces_file_prefix);
+  check_surface_volume_data<ValidPoints>(surfaces_file_prefix);
 
   if (file_system::check_if_file_exists(surfaces_file_prefix + ".h5"s)) {
     file_system::rm(surfaces_file_prefix + ".h5"s, true);
   }
 
+  // This check also uses no interpolated data
   check_ylm_data_with_greater_max_l();
 }
+
+SPECTRE_TEST_CASE(
+    "Unit.NumericalAlgorithms.Interpolator.ObserveTimeSeriesAndSurfaceData",
+    "[Unit]") {
+  domain::creators::register_derived_with_charm();
+  run_test<InterpTargetTestHelpers::ValidPoints::All>();
+  run_test<InterpTargetTestHelpers::ValidPoints::None>();
```

Comment:
Should this test test "Some"?

Reply:
This would be more difficult to test as I would need to pass in which points are invalid.  I didn't think it was worth the effort as the only difference between All and Some would be the invalid points print nan which is tested by None.  I can add a comment saying so, or put in the effort for the additional test if you would prefere

Reply:
You can just add a comment.  Same for the other test I commented on.